### PR TITLE
feat(edgeless): support resizing on edge the shape element

### PR DESCRIPTION
**Pull request by Yukiniro**
Related to #3368

### CODE REVIEW

File: packages/phasor/src/utils/math-utils.ts [hunk 1]
```diff
@@ -499,3 +499,21 @@ export function isOverlap(
     )
   );
 }
+
+export function calcPointToPointWithAngle(
```
Comment (regischen) [inline packages/phasor/src/utils/math-utils.ts:503]: there already has `rotatePoints` method in math-utils. you can use it to avoid duplication

Comment (Yukiniro) [reply packages/phasor/src/utils/math-utils.ts:503]: Ok. I will update this PR later

File: packages/blocks/src/page-block/edgeless/components/resize/resize-manager.ts [hunk 2]
```diff
@@ -210,6 +254,38 @@ export class HandleResizeManager {
       dp.x += deltaX;
       dp.y += deltaY;
 
+      if (
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/components/resize/resize-manager.ts:257]: Thanks for the contribution! But here I think "resizing on edge" refers more to following effect:

https://github.com/toeverything/blocksuite/assets/7312949/89554cbb-567c-4914-ad60-ba5d3c88cd6f

With draggable edges, we don't have to add more visible handles :)

Comment (Yukiniro) [reply packages/blocks/src/page-block/edgeless/components/resize/resize-manager.ts:257]: By the way, can I understand that the edge can be dragged in horizontal and vertical directions, but the circular handles are not displayed on the edge?

Comment (doodlewind) [reply packages/blocks/src/page-block/edgeless/components/resize/resize-manager.ts:257]: Yep!

**Review by doodlewind — APPROVED** [review]
Found a minor regression that the bottom right cursor should be `nwse-resize` instead of `ew-resize`.

But it's already quite good to land, will be great if we can expect following fixes, thanks!

**Comment by Yukiniro** [comment]
> Found a minor regression that the bottom right cursor should be `nwse-resize` instead of `ew-resize`.
> 
> But it's already quite good to land, will be great if we can expect following fixes, thanks!

Thanks for reviewing!  I will fix it later.

**Pull request merged**